test/datasets/test_amazonreviewfull.py

```diff
@@ -0,0 +1,83 @@
+import os
+import random
+import string
+import tarfile
+from collections import defaultdict
+from unittest.mock import patch
+
+from parameterized import parameterized
+from torchtext.datasets.amazonreviewfull import AmazonReviewFull
+
+from ..common.case_utils import TempDirMixin, zip_equal
+from ..common.torchtext_test_case import TorchtextTestCase
+
+
+def _get_mock_dataset(root_dir):
+    """
+    root_dir: directory to the mocked dataset
+    """
+    base_dir = os.path.join(root_dir, "AmazonReviewFull")
+    temp_dataset_dir = os.path.join(base_dir, "temp_dataset_dir")
+    os.makedirs(temp_dataset_dir, exist_ok=True)
+
+    seed = 1
+    mocked_data = defaultdict(list)
+    for file_name in ("train.csv", "test.csv"):
+        txt_file = os.path.join(temp_dataset_dir, file_name)
+        with open(txt_file, "w") as f:
+            for i in range(5):
+                label = seed % 2 + 1
```

Comment:
If you look at the readme for the AmazonReviewFull dataset, it mentions that the label for the datasets range from 1 to 5. So this line should be `label = seed % 5 + 1`